addons/account_edi/models/account_move.py

```diff
@@ -135,6 +136,169 @@ def _compute_edi_show_abandon_cancel_button(self):
+                                'base_amount_currency':     The cumulated tax base amount expressed in foreign currency.
+                                'tax_amount_currency':      The cumulated tax amount expressed in foreign currency.
+                                'records':                  A list of tax values represented as a list of dictionaries
+                                                            (see 'applied_filter' for details).
+        :return:                The full tax details for the current invoice and for each invoice line separately.
+        '''
+        self.ensure_one()
+
+        def _serialize_python_dictionary(vals):
+            return '-'.join(str(v) for v in vals.values())
+
+        def default_grouping_key(tax_values):
+            return {'tax': tax_values['tax_id']}
+
+        # Compute the taxes values for each invoice line.
+
+        invoice_lines = self.invoice_line_ids.filtered(lambda line: not line.display_type)
+        invoice_lines_tax_values_list = []
+        sign = -1 if self.is_inbound() else 1
+        for invoice_line in invoice_lines:
+            taxes_res = invoice_line.tax_ids.compute_all(
+                invoice_line.price_unit * (1 - (invoice_line.discount / 100.0)),
+                currency=invoice_line.currency_id,
+                quantity=invoice_line.quantity,
+                product=invoice_line.product_id,
+                partner=invoice_line.partner_id,
+                is_refund=invoice_line.move_id.move_type in ('in_refund', 'out_refund'),
+            )
+            rate = abs(invoice_line.balance) / abs(invoice_line.amount_currency) if invoice_line.amount_currency else 0.0
+            for tax_res in taxes_res['taxes']:
+                invoice_lines_tax_values_list.append({
+                    'base_line_id': invoice_line,
+                    'tax_id': self.env['account.tax'].browse(tax_res['id']),
+                    'tax_repartition_line_id': self.env['account.tax.repartition.line'].browse(tax_res['tax_repartition_line_id']),
+                    'base_amount': sign * invoice_line.company_currency_id.round(tax_res['base'] / rate if rate else 0.0),
+                    'tax_amount': sign * invoice_line.company_currency_id.round(tax_res['amount'] / rate if rate else 0.0),
+                    'base_amount_currency': sign * tax_res['base'],
+                    'tax_amount_currency': sign * tax_res['amount'],
+                })
+        grouping_key = grouping_key or default_grouping_key
+
+        # Apply 'applied_filter'.
+
+        if applied_filter:
+            invoice_lines_tax_values_list = [x for x in invoice_lines_tax_values_list if applied_filter(x)]
+
+        # Prepare the results.
+
+        res = {
+            'base_amount': 0.0,
+            'tax_amount': 0.0,
+            'base_amount_currency': 0.0,
+            'tax_amount_currency': 0.0,
```

Comment:
So, those are totals for the invoice, right ? You REALLY should give them another name; it's confusing. total_base_amount, total_tax_amount ?